src/main/router/create-handler-for-route.injectable.ts

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import { getInjectable } from "@ogre-tools/injectable";
+import type { ServerResponse } from "http";
+import loggerInjectable from "../../common/logger.injectable";
+import { object } from "../../common/utils";
+import type { LensApiRequest, Route } from "./route";
+import { contentTypes } from "./router-content-types";
+
+export type RouteHandler = (request: LensApiRequest<string>, response: ServerResponse) => Promise<void>;
+export type CreateHandlerForRoute = (route: Route<unknown, string>) => RouteHandler;
+
+interface LensServerResponse {
+  statusCode: number;
+  content: any;
+  headers: {
+    [name: string]: string;
+  };
+}
+
+const writeServerResponseFor = (serverResponse: ServerResponse) => ({
+  statusCode,
+  content,
+  headers,
+}: LensServerResponse) => {
+  serverResponse.statusCode = statusCode;
+
+  for (const [name, value] of object.entries(headers)) {
+    serverResponse.setHeader(name, value);
+  }
+
+  if (content instanceof Buffer) {
+    serverResponse.write(content);
+    serverResponse.end();
+  } else if (content) {
+    serverResponse.end(content);
+  } else {
+    serverResponse.end();
+  }
+};
+
+const createHandlerForRouteInjectable = getInjectable({
```

Comment:
Hmm. For sake of conversation: I wonder, in general, if instead of "creating" a "handler" with a factory-function, we could have something that "is" a handler (or handlers). That would have better chance of being reactive when required, and it would comply to Hollywood. Later for something else maybe? :)

Reply:
Fair, I mostly just moved this around because I wanted to add some logging.

This is here so that we can catch exceptions thrown accidentally and return 500s automatically as well as mapping the correct types and serialization